source/manuals/programming-languages/python/python.html.md.erb

```diff
@@ -41,6 +41,55 @@ _For more on linting (including standard settings and config) see the [linting][
 * Lint checks using [PyFlakes][]
 
 
+### Dependencies
+
+It is important to have a good strategy for specifying your project's Python
+libraries. There are two desirable characteristics:
+
+1. Reproducibility
+
+    Pin your full dependencies or you'll get unpredictability between your dev
+    environment and other environments. You want a new starter to avoid small
+    hard-to-spot problems. And you want parity between what you test locally,
+    what is tested by CI and what you deploy, or you risk new issues appearing
+    on a live server. Additionaly these things can be hard to diagnose.
+
+    Simply manually maintaining a requirements.txt doesn't capture
```

Comment:
This sentence might be better phrased as 'Manually maintaining a direct list of requirements in `requirements.txt` doesn't capture sub-dependencies, and doing that yourself is not desirable.'

Reply:
Good spot. I've tried to make it clearer by splitting it into two sentences:

    Traditionally a `requirements.txt` lists the app's immediate dependencies,
    but that leaves sub-dependencies unpinned. It makes sense to use a tool to
    generate these, but we've noticed [minor problems with using
    piptools/pip-compile and pipenv/Pipfile][dm-deps-commit]

Reply:
👍